### PR TITLE
[ONNX] Reimplement _var_mean to ensure non-negative

### PR DESCRIPTION
**Pull request by jiafatom**
The current `_var_mean` implementation cannot ensure non-negative for variance, because it is actually `E(X^2)-(E(X))^2`: numerically when the dimension number is large and X is close to 0, it can have negative numbers (like our UT shows). The new implementation is `(E(X-E(X))^2)`, it ensures non-negative because the expectation of square is non-negative for sure. 

The UT passes for the new implementation (but fails for the existing one). So it is good to go.

### CODE REVIEW

**Comment by BowenBao** [comment]
Please fix lint error.

CI test is related, looks like results are matching but there is a shape mismatch.

```
Nov 03 00:20:37 E   AssertionError: 
Nov 03 00:20:37 E   Not equal to tolerance rtol=0.001, atol=1e-07
Nov 03 00:20:37 E   
Nov 03 00:20:37 E   (shapes (1, 1, 4), (4,) mismatch)
Nov 03 00:20:37 E    x: array([[[0.692657, 0.831579, 1.032363, 1.040675]]], dtype=float32)
Nov 03 00:20:37 E    y: array([0.692657, 0.831579, 1.032363, 1.040675], dtype=float32)
```

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -2217,20 +2217,18 @@ def gather(g, self, dim, index, sparse_grad=False):
...
         mean = g.op("ReduceMean", input, axes_i=dim, keepdims_i=keepdim)
         redudced_dims = g.op("Shape", input)
         # dim could contain one or multiple dimensions
         redudced_dims = g.op("Gather", redudced_dims, g.op("Constant", value_t=torch.tensor(dim)), axis_i=0)
         num_elements = g.op("ReduceProd", redudced_dims, keepdims_i=0)
-    meansqrd = g.op("Mul", mean, mean)
-    var = g.op("Sub", sqrdmean, meansqrd)
+    sub_v = g.op("Sub", input, mean)
+    sqr_sub = g.op("Mul", sub_v, sub_v)
+    var = g.op("ReduceMean", sqr_sub, axes_i=dim, keepdims_i=0)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2231]: error may from here `keepdims_i`

Comment (jiafatom) [reply torch/onnx/symbolic_opset9.py:2231]: Thanks, yes, should be `keepdims_i=keepdims`, running tests now.

**Comment by KsenijaS** [comment]
looks good, please check caffe2 test falures

**Comment by jiafatom** [comment]
> looks good, please check caffe2 test falures

fixed. Thank you.

**Review by BowenBao — APPROVED** [review]
LGTM, thanks!

**Review by bzinodev — APPROVED** [review]

**Pull request closed without merging**